var/spack/repos/builtin/packages/espresso/package.py

```diff
@@ -91,38 +101,53 @@ def install(self, spec, prefix):
             options.append('--enable-openmp')
 
         if '+scalapack' in spec:
-            options.append('--with-scalapack=yes')
+            scalapack_option = 'intel' if '^intel-mkl' in spec else 'yes'
+            options.append('--with-scalapack={0}'.format(scalapack_option))
 
         if '+elpa' in spec:
-            if spec.satisfies('@:5.4.0'):
-                options.append('--with-elpa=yes')
-            else:
-                options.extend([
-                    '--with-elpa-include={0}'.format(
-                        join_path(spec['elpa'].prefix,
-                                  'include',
-                                  'elpa-{0}'.format(str(spec['elpa'].version)),
-                                  'modules')
-                    ),
-                    '--with-elpa-lib={0}'.format(spec['elpa'].libs.joined())
-                ])
+
+            # Spec for elpa
+            elpa = spec['elpa']
+
+            # Find where the Fortran module resides
+            elpa_module = find(elpa.prefix, 'elpa.mod')
+
+            # Compute the include directory from there: versions
+            # of espresso prior to 6.1 requires -I in front of the directory
+            elpa_include = '' if '@6.1:' in spec else '-I'
+            elpa_include += os.path.dirname(elpa_module[0])
+
+            options.extend([
+                '--with-elpa-include={0}'.format(elpa_include),
+                '--with-elpa-lib={0}'.format(elpa.libs[0])
+            ])
+
+        if '+hdf5' in spec:
+            options.append('--with-hdf5={0}'.format(spec['hdf5'].prefix))
 
         # Add a list of directories to search
         search_list = []
         for dependency_spec in spec.dependencies():
-            search_list.extend([dependency_spec.prefix.lib,
-                                dependency_spec.prefix.lib64])
+            search_list.extend([
+                dependency_spec.prefix.lib,
+                dependency_spec.prefix.lib64
+            ])
 
         search_list = " ".join(search_list)
-        options.append('LIBDIRS=%s' % search_list)
-        options.append('F90=%s' % os.environ['FC'])
+
+        options.extend([
+            'LIBDIRS={0}'.format(search_list),
+            'F90={0}'.format(env['SPACK_FC']),
+            'CC={0}'.format(env['SPACK_CC'])
+        ])
 
         configure(*options)
+
         make('all')
 
```

Comment:
Not accodring to [this page](http://www.quantum-espresso.org/wp-content/uploads/Doc/user_guide/node12.html)